langchain4j/src/main/java/dev/langchain4j/memory/chat/ChatMemoryService.java

```diff
@@ -17,19 +18,20 @@ public class ChatMemoryService {
     private ChatMemoryProvider chatMemoryProvider;
 
     public ChatMemoryService(ChatMemoryProvider chatMemoryProvider) {
-        this.chatMemories = Collections.synchronizedMap(new WeakHashMap<>());
+        this.chatMemories = new ConcurrentHashMap<>();
```

Comment:
Since the user is now allowed to explicitly clear the `ChatMemory`, I turned this back into a `ConcurrentHashMap`.